standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -8373,6 +8374,11 @@ public void commit_txn(CommitTxnRequest rqst) throws TException {
       if (listeners != null && !listeners.isEmpty()) {
         MetaStoreListenerNotifier.notifyEvent(listeners, EventType.COMMIT_TXN,
                 new CommitTxnEvent(rqst.getTxnid(), this));
+        CompactionInfo compactionInfo = getTxnHandler().getCompactionByTxnId(rqst.getTxnid());
```

Comment:
I think this overlap with code in CompactionTxnHandler

Reply:
This does a little bit different thing. In TxnHandler we notify the transactional eventListeners using the same jdbc connection from the txnhandler. The event will be committed together with the "hive transaction commit". This on the other hand notifies the "normal" event listeners. The file metadata cache strictly needs only the first one, but every other transactional event is fired as normal event also, so I don't want to break this pattern.